src/main/java/ch/njol/skript/lang/function/Namespace.java

```diff
@@ -92,38 +92,93 @@ public boolean equals(@Nullable Object obj) {
 			return true;
 		}
 	}
+
+	/**
+	 * The key used in the signature and function maps
+	 */
+	public static class Info {
+
+		/**
+		 * Name of the function
+		 */
+		private final String name;
+
+		/**
+		 * Whether the function is local
+		 */
+		private final boolean local;
+
+		public Info(String name, boolean local) {
+			this.name = name;
+			this.local = local;
+		}
+
+		public String getName() {
+			return name;
+		}
+
+		public boolean isLocal() {
+			return local;
+		}
+
+		@Override
+		public int hashCode() {
+			int result = getName().hashCode();
+			result = 31 * result + (isLocal() ? 1 : 0);
+			return result;
+		}
+
+		@Override
+		public boolean equals(Object o) {
+			if (this == o)
+				return true;
+			if (!(o instanceof Info))
+				return false;
+
+			Info info = (Info) o;
+
+			if (isLocal() != info.isLocal())
+				return false;
+			return getName().equals(info.getName());
+		}
+	}
 	
 	/**
 	 * Signatures of known functions.
 	 */
-	private final Map<String, Signature<?>> signatures;
-	
+	private final Map<Info, Signature<?>> signatures;
+
 	/**
 	 * Known functions. Populated as function bodies are loaded.
 	 */
-	private final Map<String, Function<?>> functions;
-	
+	private final Map<Info, Function<?>> functions;
+
 	public Namespace() {
 		this.signatures = new HashMap<>();
 		this.functions = new HashMap<>();
 	}
 	
+	@Nullable
+	public Signature<?> getSignature(String name, boolean local) {
+		return signatures.get(new Info(name, local));
+	}
+
 	@Nullable
 	public Signature<?> getSignature(String name) {
-		return signatures.get(name);
+		Signature<?> signature = getSignature(name, true);
+		return signature == null ? getSignature(name, false) : signature;
 	}
-	
+
 	public void addSignature(Signature<?> sign) {
-		if (signatures.containsKey(sign.getName())) {
+		if (signatures.containsKey(new Info(sign.getName(), sign.local)))
```

Comment:
We can just make a variable for Info in this method

Reply:
like a parameter?

Reply:
In the method you can add it, as you are creating `Info` twice with the same signature information
ex:
```java
Info info = new Info(sign.getName(), sign.local);
```
and just reuse it :)